modules/apps/29-fee/ibc_middleware.go

```diff
@@ -168,6 +168,14 @@ func (im IBCMiddleware) OnChanCloseConfirm(
 		return err
 	}
 
+	if im.keeper.IsLocked(ctx) {
+		return types.ErrFeeModuleLocked
+	}
+
+	if !im.keeper.IsFeeEnabled(ctx, portID, channelID) {
+		return types.ErrFeeNotEnabled
+	}
```

Comment:
It is good for us to think about why we even have "fee enabled channels". This is because non fee enabled channels still use the same middleware stack as fee enabled channels. Thus this check prevents closing of a non fee enabled channel. A key requirement of the middleware design is that middleware not enabled for a specific channel must not affect the behaviour of that channel (this requirement is broken by returning an error). 

We should:
- return nil if the fee is not enabled (exit early)
- move this check above `IsLocked`

The reasoning for the second point is that we don't want non fee enabled channels to get affected by the fee module getting locked. A locked fee module should not affect non fee enabled channels

Reply:
Thanks you. It helped me understand. I'll make this change.